snuba/subscriptions/scheduler.py

```diff
@@ -61,6 +308,11 @@ def find(
             math.ceil(interval.upper.timestamp()),
         ):
             for subscription in subscriptions:
-                resolution = int(subscription.data.resolution.total_seconds())
-                if timestamp % resolution == 0:
-                    yield ScheduledTask(datetime.fromtimestamp(timestamp), subscription)
+                task = self.__builder.get_task(subscription, timestamp)
+                if task is not None:
+                    yield task
+
+        metrics = self.__builder.reset_metrics()
+        if any(metric for metric in metrics if metric[1] > 0):
+            for metric in metrics:
+                self.__metrics.increment(metric[0], metric[1], tags=metric[2])
```

Comment:
Isn't this a complex way to just increment a metric from each builder or is there another reason for it being done like this? Can't you just call `metrics.increment()` inside each task builder instead of passing that metrics state around and then having extra logic to reset it?

Reply:
YEs, there is a reason for not doing it inside the builder itself. It is to reduce the amount of metrics we record by batching them. Otherwise we would be potentially calling it 300 times per partition. This way we do one call per  time range. 
Each call is a network call which can contribute to the backlog.

Reply:
I don't mean to log it every time you increment count, just change the `reset_metrics` function to do `record_metrics` and trigger it from here.